packages/vc-verification/src/models/cache-interfaces.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  IRoleDefinitionV2,
+  RoleCredentialSubject,
+} from '@energyweb/credential-governance';
+import { VerifiableCredential } from '@ew-did-registry/credentials-interface';
+import { RoleEIP191JWT } from './credentials-types';
+import { IDIDDocument } from '@ew-did-registry/did-resolver-interface';
+
+/**
+ * An Interface to cache DID Document from Blockchain and SSI-Hub
+ */
+export interface IDIDDocumentCache {
+  /**
+   * Stores DID Document
+   * @param did user DID
+   * @param data DIDDocument of the user (DID)
+   */
+  setDIDDocument(did: string, data: IDIDDocument): void;
+
+  /**
+   * Returns cached DID Document (services)
+   * @param did user DID
+   */
+  getDIDDocument(did: string): IDIDDocument | undefined;
+}
+
+/**
+ * An Interface to cache role credential from Blockchain and SSI-Hub
```

Comment:
Same rationale as previous comment
```suggestion
 * An Interface to cache role credential
```